src/components/simplesearchengine/SimpleSearchEngine1L.java

```diff
@@ -0,0 +1,706 @@
+package components.simplesearchengine;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import components.map.Map;
+import components.map.Map.Pair;
+import components.map.Map4;
+import components.sequence.Sequence;
+import components.sequence.Sequence3;
+import components.tree.Tree;
+import components.tree.Tree1;
+
+/**
+ * {@code SimpleSearchEngine} represented as {@link components.map.Map4} and
+ * trie on {@link components.tree.Tree1} with implementations of primary
+ * methods.
+ *
+ * @param <T>
+ *            type of {@code SimpleSearchEngine} entries
+ *
+ * @mathsubtypes <pre>
+ * SIMPLE_SEARCH_ENGINE_1L_MODEL is (
+ *   insertion_mode: boolean,
+ *   entries: finite map of type (String, T),
+ *   searchStructure: trie containing sorted letters of type String in entries (String, T)
+ *  )
+ * </pre>
+ * @mathmodel type SimpleSearchEngine1L<T> is modeled by
+ *            SIMPLE_SEARCH_ENGINE_1L_MODEL
+ *
+ * @author Saurav Poudyel
+ *
+ */
+public class SimpleSearchEngine1L<T> extends SimpleSearchEngineSecondary<T> {
```

Comment:
Heads up that this looks excellent. Way above and beyond what I would expect for this.